### PR TITLE
GasFilterComponent

### PR DESCRIPTION
**Pull request by collinlunn**
Gas filter, sprites and visualizer not included.

### CODE REVIEW

File: Content.Server/GameObjects/Components/Atmos/Piping/GasFilterComponent.cs [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+            serializer.DataField(ref _initialFilterOutletDirection, "filterOutletDirection", PipeDirection.None);
+            serializer.DataField(ref _initialOutletDirection, "outletDirection", PipeDirection.None);
+        }
+
+        public override void Initialize()
+        {
+            base.Initialize();
+            Owner.EnsureComponent<PipeNetDeviceComponent>();
+            SetPipes();
+            Owner.TryGetComponent(out _appearance);
```
Comment (PaulRitter) [inline Content.Server/GameObjects/Components/Atmos/Piping/GasFilterComponent.cs:108]: use componentdependencies here

File: Content.Shared/Atmos/Atmospherics.cs [hunk 2]
```diff
@@ -246,6 +248,7 @@ static Atmospherics()
     /// <summary>
     ///     Gases to Ids. Keep these updated with the prototypes!
     /// </summary>
+    [Serializable, NetSerializable]
```
Comment (PaulRitter) [inline Content.Shared/Atmos/Atmospherics.cs:251]: why is this needed?

**Review by PaulRitter — CHANGES_REQUESTED** [review]
cannot evaluate the atmos parts since i haven't worked with that yet, otherwise just some nitpicks

File: Content.Shared/Atmos/Atmospherics.cs (hunk 2 above)
Comment (collinlunn) [reply Content.Shared/Atmos/Atmospherics.cs:251]: Needed for Gas types to be used by VV

Comment (PaulRitter) [reply Content.Shared/Atmos/Atmospherics.cs:251]: ah, makes sense

**Review by PaulRitter — APPROVED** [review]

**Pull request merged**